src/libs/Fullstory/index.ts

```diff
@@ -91,5 +101,86 @@ const FS = {
     init: (_value: OnyxEntry<UserMetadata>) => {},
 };
 
+/**
+ * Extract values from non-scraped at build time attribute WEB_PROP_ATTR,
+ * reevaluate "fs-class".
+ */
+function parseFSAttributes(): void {
+    window?.document?.querySelectorAll(`[${WEB_PROP_ATTR}]`).forEach((o) => {
+        const attr = o.getAttribute(WEB_PROP_ATTR) ?? '';
+        if (!/fs-/gim.test(attr)) {
+            return;
+        }
+
+        const fsAttrs = attr.match(/fs-[a-zA-Z0-9_-]+/g) ?? [];
+        o.setAttribute('fs-class', fsAttrs.join(','));
+
+        let cleanedAttrs = attr;
+        fsAttrs.forEach((fsAttr) => {
+            cleanedAttrs = cleanedAttrs.replace(fsAttr, '');
+        });
+
+        cleanedAttrs = cleanedAttrs
+            .replace(/,+/g, ',')
+            .replace(/\s*,\s*/g, ',')
+            .replace(/^,+|,+$/g, '')
+            .replace(/\s+/g, ' ')
+            .trim();
+
+        if (cleanedAttrs) {
+            o.setAttribute(WEB_PROP_ATTR, cleanedAttrs);
+        } else {
+            o.removeAttribute(WEB_PROP_ATTR);
+        }
+    });
+}
+
+/*
+    prefix? if component name should be used as a prefix,
+    in case data-test-id attribute usage,
+    clean component name should be preserved in data-test-id.
+*/
+function getFSAttributes(name: string, mask: boolean, prefix: boolean): string {
+    const componentPrefix = prefix ? `${name},` : '';
+    const componentSuffix = name ? `,fs-${name}` : '';
+    const fsAttrValue = `${componentPrefix}${mask ? MASK : UNMASK}${componentSuffix}`;
+    /*
+    testID: componentName,fs-unmask,fs-componentName
+    fsClass: fs-unmask,fs-componentName
+    */
+    return fsAttrValue;
+}
+
+function getChatFSAttributes(context: OnyxEntry<PersonalDetailsList>, name: string, report: OnyxInputOrEntry<Report>, prefix: boolean): string {
+    let componentPrefix = prefix ? `${name},` : '';
+    let componentSuffix = name ? `,fs-${name}` : '';
+    let fsAttrValue = '';
+
+    if (isConciergeChatReport(report)) {
+        componentPrefix = prefix ? `${CONCIERGE}-${name},` : '';
+        componentSuffix = name ? `,fs-${name}` : '';
+        /*
+        concierge-chatMessage,fs-unmask,fs-chatMessage
+        */
+        fsAttrValue = `${componentPrefix}${UNMASK}${componentSuffix}`;
+    } else if (isExpensifyAndCustomerChat(context, report)) {
+        componentPrefix = prefix ? `${CUSTOMER}-${name},` : '';
+        componentSuffix = name ? `,fs-${name}` : '';
+        /*
+        customer-chatMessage,fs-unmask,fs-chatMessage
+        */
+        fsAttrValue = `${componentPrefix}${UNMASK}${componentSuffix}`;
+    } else {
+        componentPrefix = prefix ? `${OTHER}-${name},` : '';
+        componentSuffix = name ? `,fs-${name}` : '';
+        /*
+        other-chatMessage,fs-mask,fs-chatMessage
+        */
+        fsAttrValue = `${componentPrefix}${MASK}${componentSuffix}`;
+    }
+
+    return fsAttrValue;
+}
```

Comment:
I think this logic can be simplified as 
```suggestion
function getChatFSAttributes(context: OnyxEntry<PersonalDetailsList>, name: string, report: OnyxInputOrEntry<Report>, prefix: boolean): string {
    const baseName = name ? `,fs-${name}` : '';
    let componentPrefix = '';
    let maskType = MASK;

    if (isConciergeChatReport(report)) {
        componentPrefix = CONCIERGE;
        maskType = UNMASK;
    } else if (isExpensifyAndCustomerChat(context, report)) {
        componentPrefix = CUSTOMER;
        maskType = UNMASK;
    } else {
        componentPrefix = OTHER;
    }

    const prefixPart = prefix ? `${componentPrefix}-${name},` : '';
    return `${prefixPart}${maskType}${baseName}`;
}
```